### ISSUE TITLE
Nest thermostat own temperature entity

### ISSUE THREAD

**Issue by iku777** [iMicknl/ha-nest-protect#263]
### Description

I am using v0.4.0b2 and I must say it works great. I have both Protect and 4 sensors and all is very smooth. My request would be to also get the temperature readings from the Nest Thermostat itself. I also have the Google Nest integration but the temperature it reports comes from the sensor you are monitoring at that time (in my case one of my sensor), so I have no reliable way to way the temperature at the thermostat itself. 

I assume given you already have the sensors temperature, you must also get the temperature of the thermostat itself in the feed you get.

Thanks and keep up the great work

**Comment by iMicknl** [comment]
Can you add your diagnostics?

**Comment by iku777** [comment]
Sure.
Are you refering to this?

[config_entry-nest_protect-75db7a8d5445b30732b369f64fe4eedb.json.txt](https://github.com/iMicknl/ha-nest-protect/files/12749619/config_entry-nest_protect-75db7a8d5445b30732b369f64fe4eedb.json.txt)

**Comment by iku777** [comment]
Just checking, did you find anything interresting in that diagnostics log? @iMicknl

**Comment by c0dyJK** [comment]
I see my Nest thermostat’s temperature sensor entity, as well as my wireless Nest temp sensor. Are you wishing to be able to isolate the temperature sensor that is built into the thermostat as its own entity (i.e. not grouped in a dashboard with your thermostat card?)

**Comment by iku777** [comment]
Hi. There are 2 scenarios. 
1-You use the thermostat's own temperature sensor to control the temperature and the additional nest sensor (puck) as an additional piece of information. In that case, everything is fine.
2-You use a sensor's temperature (puck) to control the thermostat (which is my case). In that case both the thermostat's temperature and the sensor temperature (puck) show up the same in HA. What I would like is to be able to still see the thermostat's own temperature in HA as a separate entity to not 'loose' that info.
Please let me know if clear, else can put some pictures to illustrate

**Comment by c0dyJK** [comment]
That definitely clarifies it and it aligns exactly with what I was researching to be able to do as well (scenario 2)

**Comment by iMicknl** [comment]
@iku777 I just had a look and it is not part of your diagnostics. Why would you need this, if you can get this from the Nest integration in core?

**Comment by c0dyJK** [comment]
I just double checked to clarify further on my end. I checked the default Overview dashboard which has all new integrations added (my dashboards only have the Nest card showing) I see the default thermostat card, and then separate entities for Living Room (nest thermostat) that shows up as just a temp sensor in HA and says 68.7 degrees and Bedroom (temp sensor) which reads 70.2 degrees. I set the bedroom as the “host” in the Google nest app and my HA thermostat app is only showing the “hosts” temp as it overrides the thermostat one. However, as mentioned, in the default overview which shows all cards and entities I was able to still see both sensor readings.

**Comment by iku777** [comment]
@iMicknl Thanks for the follow-up.
Right now my thermostat (Hallway in my case) temperature is nowhere to be found in HA.
I have attached some screenshots to illustrate
1-In Nest app, I have Hallway and 4 sensors. My thermostat is setup to use C's bedroom to trigger heating.
![Nest](https://github.com/iMicknl/ha-nest-protect/assets/82530487/ba63d3de-3ecf-459e-b52d-40fa7f7ad66f)
2-In the Nest core HA integration, you can see the temperature reported is the same as C's bedroom, NOT the Hallway
![HA-Nest](https://github.com/iMicknl/ha-nest-protect/assets/82530487/9e708d3a-6b74-4832-a3e3-b95e86262c3a)
3-In the Nest protect HA integration, I see all my sensors, but not the Hallway.
![HA-Nest protect](https://github.com/iMicknl/ha-nest-protect/assets/82530487/ade2b468-a365-4e8f-8e5c-5a15583f738a)
So the temperature of the hallway itself (23 in that case) is nowhere to be found in HA, which is what I would like to see and use for other purposes.

@c0dyJK Are you saying you found a way that, when your host in Nest app is set to bedroom, you still see your living room temp in HA? If that is the case, would you mind sharing some snapshots how you got that?

**Comment by c0dyJK** [comment]
@iku777 Unfortunately the default “control” thermostat card for the Nest does act the way you are describing: by showing only the temperature of the current selected sensor or ‘host’ and thus overriding the Nest thermostat’s temperature reading for that room. 

I am, however, still able to see the Nest thermostat as a separate temperature sensor entity in HA which I believe is what iMicknl was alluding to when saying you can get the info from the Nest integration in core (as opposed to from Nest Protect integration for the room sensors)

I am realizing I’ll need to make a custom card to group them all together visually in the dashboard. No doubt will be worse looking than the default Thermostat card but I’ve seen some decent looking templates I could use - I’ll share if I find a solution before anyone else engaged

**Comment by iku777** [comment]
@c0dyJK 
Ok, I think I understand what you mean.
So in the Nest integration you both see a Climate entity (showing temp of 71) and a sensor (showing temp of 70.9). 
Basically both of those are showing the same thing (which is your host in Nest, Bedroom), but right now there is nowhere in HA where you can see the Thermostat temperature (Living room), which is at 70. Is that correct?

**Comment by puterboy** [comment]
I have the same request.
Basically, when the Nest Thermostat is set to use the thermostat itself as the temperature sensor then all is good:

1. Google Nest integration entity shows the temperature at the thermostat
2. Nest Protect entity shows the temperature at the remote sensor

However, when the Nest Thermostat is set to use the Remote sensor as the temperature sensor then:
1. Google Nest integration entity shows the temperature at the remote sensor
2. Nest Protect entity shows the temperature at the remote sensor

So you lose the temperature at the Thermostat itself.

The solution could be to add an entity for the Nest Thermostat itself to Nest Protect.

**Comment by adamk33n3r** [comment]
We generally keep our child's room's puck as what controls the temperature, so ya I'm not able to see what the temperature is at the Nest itself. Would really love to see this added somehow, to either core or this project.

**Comment by puterboy** [comment]
Is anyone looking at this?
To summarize the problem:
1. The standard Nest Integration creates a temperature entity that corresponds to the *active* sensor for that thermostat (either the thermostat or the remote puck)
2. Nest Protect creates an entity that *always* tracks the remote puck

So, if the thermostat is set to use the remote puck as a source then you have two **redundant** entities tracking the remote puck and **zero** entities tracking the thermostat.
Even worse potentially if you frequently switch between thermostat and puck then the temperature history for the Nest integration is inconsistent since its source is really alternating between different sensors.

So the request here is whether it is possible to **add** an additional temperature sensor entity to Nest Protect that is dedicated to tracking the temperature at the thermostat. Then one would potentially have 3 temperature entities for each thermostat:
1. Nest Protect entity always tracking the thermostat temperature
2. Nest Protect entity always tracking the puck temperature
3. Nest Integration entity tracking whatever sensor the thermostat is set to use

This would be perfect for all the use cases noted in this thread.

@iMicknl - I am willing to help troubleshoot or modify code here if you can share some pointers of where to look...

**Comment by puterboy** [comment]
BTW, the `homebridge-nest` integration for Homebridge seems to be able to read date from the thermostat.
See https://github.com/chrisjshull/homebridge-nest/tree/789799ce5d926345c96aa9839c66158a42d23e76 and in particular the file https://github.com/chrisjshull/homebridge-nest/blob/master/lib/nest-thermostat-accessory.js

It seems like while `kryptonite` is the keyword for a remote puck, `zirconium` is the keyword for a thermostat.

**Comment by iMicknl** [comment]
They are using the Protobuf implementation (`this.isZirconium = (this.device.protobuf_device_type == 'google.resource.GoogleZirconium1Resource');` which is not implemented in this HA integration.

**Comment by puterboy** [comment]
Would that be hard to implement?
Or can one just adapt similar code to what is done for `kryptonite`?

**Comment by iMicknl** [comment]
No, the other code in this integration does use the REST API and not the protobuf API. This will be hard to implement and I won't be working on this myself.

**Comment by puterboy** [comment]
Thanks.
Ahhh then this is probably way beyond my abilities to adapt.
I guess I will try using the Homebridge plug in as suggested in `https://www.reddit.com/r/homeassistant/comments/rhvbsm/is_there_any_support_for_the_nest_temperature/`, unless you or others have any other suggestions.
Apparently the Homebridge nest integration works with any nest device.

**Comment by puterboy** [comment]
It seems like the Nest Legacy integration is able to do all this and more.
I will check out that solution as it seems to promise support for more sensors (and credits your integration as well as Nest-Homebridge for inspiration)

**Comment by tronikos** [comment]
For the record, the integration mentioned in the last comment is https://github.com/tronikos/nest_legacy
It fully supports Nest thermostats.

**Comment by iku777** [comment]
I confirm that with tronikos integration, my thermostat has an additional attribute available (mine was called "Backplate temperature") which displays the thermostat's own temperature, even though the thermostat is using another Google sensor to control temperature.
